examples/hello_world/proto/BUILD

```diff
@@ -33,10 +35,22 @@ cc_proto_library(
     deps = [":hello_world_proto"],
 )
 
+java_proto_library(
+    name = "hello_world_java_proto",
+    deps = [":hello_world_proto"],
+)
+
 cc_grpc_library(
     name = "hello_world_cc_grpc",
     srcs = ["hello_world_proto"],
     grpc_only = True,
     well_known_protos = True,
     deps = ["hello_world_cc_proto"],
 )
+
+java_grpc_library(
+    name = "hello_world_java_grpc",
+    srcs = ["hello_world_proto"],
+    flavor = "lite",  # Lite version is optimized for mobile devices.
+    deps = ["hello_world_java_proto"],
```

Comment:
This should include constraints = ["android"]. This ensures that the gRPC library does not depend on any non-lite java proto libraries. This should also use the lite versions to be added above.